SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -298,10 +354,90 @@ void UGlobalStateManager::AuthorityChanged(bool bWorkerAuthority, Worker_EntityI
 	{
 		// Make sure we update our known entity id for the GSM when we receive authority.
 		GlobalStateManagerEntityId = CurrentEntityID;
+
+		SetCanBeginPlay(true);
+		BecomeAuthoritativeOverAllActors();
+		TriggerBeginPlay();
+
+		// Start accepting players only AFTER we've triggered BeginPlay
 		SetAcceptingPlayers(true);
 	}
 }
 
+void UGlobalStateManager::BeginDestroy()
+{
+	Super::BeginDestroy();
+
+#ifdef WITH_EDITOR
+	if (NetDriver->StaticComponentView->HasAuthority(GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID))
+	{
+		// If we are deleting dynamically spawned entities, we need to
+		if (GetDefault<ULevelEditorPlaySettings>()->IsDeleteDynamicEntitiesActive())
+		{
+			// Reset the BeginPlay flag so Startup Actors are properly managed.
+			SetCanBeginPlay(false);
```

Comment:
Shouldn't we do this regardless of dynamic deletion of entities?

Reply:
No, because if BeginPlay was already called in the world, startup actors have already had beginplay called on them.